src/WinRT.Runtime/Interop/IActivationFactory.cs

```diff
@@ -70,13 +71,45 @@ public static unsafe IObjectReference ActivateInstanceUnsafe(IObjectReference ob
 
             try
             {
-                return ComWrappersSupport.GetObjectReferenceForInterface<IUnknownVftbl>(instancePtr, global::WinRT.Interop.IID.IID_IUnknown);
+                return ComWrappersSupport.GetObjectReferenceForInterface<IUnknownVftbl>(instancePtr, global::WinRT.Interop.IID.IID_IInspectable, false);
             }
             finally
             {
                 MarshalInspectable<object>.DisposeAbi(instancePtr);
             }
-        }
+        }
+#else
+        /// <summary>
+        /// Activates an instance from a given activation factory <see cref="IObjectReference"/> instance with the result
+        /// pointing to the interface for the given iid.
+        /// </summary>
+        /// <param name="objectReference">The input activation factory <see cref="IObjectReference"/> instance to use.</param>
+        /// <param name="iid">The iid for the interface which the resulting <see cref="IObjectReference"/> will be for.</param>
+        /// <returns>The resulting <see cref="IObjectReference"/> instance created from <paramref name="objectReference"/> and QI to the given <paramref name="iid"/>.</returns>
+        /// <remarks>
+        /// <para>This method assumes <paramref name="objectReference"/> is wrapping an <c>IActivationFactory</c> instance (with no validation).</para>
+        /// <para>This method is only meant to be used by the generated projections, and not by consumers of CsWinRT directly.</para>
+        /// </remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static unsafe IObjectReference ActivateInstanceUnsafe(IObjectReference objectReference, Guid iid)
+        {
+            IntPtr thisPtr = objectReference.ThisPtr;
+            IntPtr instancePtr;
+
+            ExceptionHelpers.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, IntPtr*, int>**)thisPtr)[6](thisPtr, &instancePtr));
+
+            GC.KeepAlive(objectReference);
+
+            try
+            {
+                return ComWrappersSupport.GetObjectReferenceForInterface<IUnknownVftbl>(instancePtr, iid, true);
```

Comment:
Just so I understand: how come we're always requiring QI on .NET, but not downlevel? Can we never skip it here?

Reply:
In both scenarios, there is a QI given we get `IInspectable`.  We are just optimizing here by doing it earlier.  On !NET, we will have a QI done right after this when it goes into the default interface constructor.  I just had an opportunity to reduce an object allocation on NET by doing the QI here and thereby simplifying the code in the consumer.